src/rgw/rgw_bucket.cc

```diff
@@ -1175,18 +1177,24 @@ int RGWBucketAdminOp::limit_check(rgw::sal::Store* store,
 				     * the loop body */
 
 	ret = bucket->get_bucket_info(dpp, null_yield);
-	if (ret < 0)
+	if (ret < 0) {
 	  continue;
+	}
 
 	/* need stats for num_entries */
 	string bucket_ver, master_ver;
 	std::map<RGWObjCategory, RGWStorageStats> stats;
-	const auto& latest_log = bucket->get_info().layout.logs.back();
-	const auto& index = log_to_index_layout(latest_log);
-	ret = bucket->get_bucket_stats(dpp, index, RGW_NO_SHARD, &bucket_ver, &master_ver, stats, nullptr);
+	const auto index = bucket->get_info().last_index_layout_gen_o();
+	if (!index) {
+	  ldpp_dout(dpp, 0) << "ERROR: " << __func__ <<
+	    ": tried to access index of indexless bucket " << bucket->get_name() << dendl;
+	  return -EINVAL;
+	}
+	ret = bucket->get_bucket_stats(dpp, *index, RGW_NO_SHARD, &bucket_ver, &master_ver, stats, nullptr);
```

Comment:
in these bucket stats cases, we can just use `layout.current_index` directly; we don't need to rely on `logs.back()` and `log_to_index_layout()` here. sorry that i didn't catch this during initial review

i made this change in https://github.com/ceph/ceph/pull/44663